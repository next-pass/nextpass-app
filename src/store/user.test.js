import reducer, {
  loginAct,
  logOutAct,
  updateAct
} from './user';

let state = undefined;

it('1- default', () => {
  expect(reducer(state, {})).toMatchSnapshot();
});


it('2- login', () => {
  const act = loginAct({username: 'username', image: 'https://foo.com/bar.jpg'});
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('3- update', () => {
  const act = updateAct({createdAt: 121331});
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});


it('4- logout', () => {
  const act = logOutAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});
