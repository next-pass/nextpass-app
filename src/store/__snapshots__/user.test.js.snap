// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default 1`] = `null`;

exports[`2- login 1`] = `
Object {
  "image": "https://foo.com/bar.jpg",
  "username": "username",
}
`;

exports[`3- update 1`] = `
Object {
  "createdAt": 121331,
  "image": "https://foo.com/bar.jpg",
  "username": "username",
}
`;

exports[`4- logout 1`] = `null`;
